### PR TITLE
Metrics

### PR DESCRIPTION
**Pull request by Paanugoban**
This adds a few functions to be able to do metrics found in GIT (Bleu, Rouge, Meteor and CIDER-D). I found many implementations of these models, but I used the natural language toolkit that comes with python and a brand new package on python that you would have to install manually, https://pypi.org/project/aac-metrics/. The template in which you would ask these functions can be found in the comments in the function or test function below. Preliminary testing seems to suggest it works, but if it doesn't, we can also adjust to use hugging face evaluate and use another package that outputs (Bleu and Rouge). However for CIDER-D its difficult to find alternatives. The implementation of all these manually seemed complex, that I thought it's better to work off something that was implemented to ensure its working.

### CODE REVIEW

File: src/metrics.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+'''
+This function calculates the BLEU-4 score for an entire corpus. That means we would have a list of all the captions in references and a list of all the predicted captions in candidates.
+'''
+def calculate_bleu_score_corpus(references, candidates):    
+    #Example: [[ref1a sentence, ref1bsentence], [ref2a sentence, ref2b sentence]]
+    #[['The cat is on the mat', 'there is a cat on the wall'],['The cat is not on the mat', 'there is a dog on the wall']]
+
+    #Example: [cand1 sentence, cand2 sentence]
+    #['The cat is on the mat','The cat is not on the mat']
```
Comment (farazali7) [inline src/metrics.py]: Try to follow standard documentation format for function docs, like this for ex: https://github.com/farazali7/multimodal-xray/blob/99a267faf6980217b7308021e09c6aa9441d06e3/src/models/attention.py#L11

File: src/metrics.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+'''
+Using AAC-metrics package to output cider-d score
+'''
+def calculate_cider_d_score(references,candidates):
+    try:
+        # Using shell=True to run the command through the shell
+        result = subprocess.run("aac-metrics-download", shell=True, check=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE, text=True)
+        print("Output:", result.stdout)
+    except subprocess.CalledProcessError as e:
+        print("Error occurred:", e.stderr)
```
Comment (farazali7) [inline src/metrics.py:168]: This is a bit awkward... any alternatives for CIDEr score without having to invoke the shell via this package?

File: src/metrics.py [hunk 3]
```diff
@@ -0,0 +1,148 @@
...
+
+def test_metrics():
+    candidate= ["a man is speaking", "rain falls"]
+    reference = [["a man speaks.", "someone speaks.", "a man is speaking while a bird is chirping in the background"], ["rain is falling hard on a surface"]]
+    #reference = ['this is small test']
+    #candidate = ['this is a test']
+    print(calculate_cider_d_score(reference, candidate))
+
+
+#test_metrics()
```
Comment (farazali7) [inline src/metrics.py]: NP: Create a tests/ folder outside of src (in root project directory) and move this function inside of a test_metrics.py file.

**Review by farazali7 — CHANGES_REQUESTED** [review]
Good work. Just some small change requests before we merge this in.

File: src/metrics.py (hunk 2 above)
Comment (farazali7) [reply src/metrics.py:168]: If not or seems like a lot of effort right now, then let's just revisit this metric a little later.

**Comment by Paanugoban** [comment]
@farazali7 I have the suggested changes to the code

**Review by farazali7 — APPROVED** [review]

**Pull request merged**